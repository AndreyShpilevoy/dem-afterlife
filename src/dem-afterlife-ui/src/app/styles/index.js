import ThemedStyleSheet from 'react-with-styles/lib/ThemedStyleSheet';
import {css, withStyles, ThemeProvider} from 'react-with-styles';
import jssInterface from './jssInterface';
import defaultTheme from './themes/default';
import overriddenTheme from './themes/overridden';

ThemedStyleSheet.registerDefaultTheme(defaultTheme);
ThemedStyleSheet.registerTheme('overridden', overriddenTheme);
ThemedStyleSheet.registerInterface(jssInterface);

const getClassNameFromCss = styles => css(styles).className;

export {getClassNameFromCss as css, withStyles, ThemeProvider, ThemedStyleSheet};