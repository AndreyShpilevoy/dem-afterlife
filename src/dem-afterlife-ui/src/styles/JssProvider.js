import React from "react";
import { node } from "prop-types";
import { create } from "jss";
import jssGlobal from "jss-global";
import jssExtend from "jss-extend";
import jssNested from "jss-nested";
import jssCamelCase from "jss-camel-case";
import jssDefaultUnit from "jss-default-unit";
import jssExpand from "jss-expand";
import jssVendorPrefixer from "jss-vendor-prefixer";
import jssPropsSort from "jss-props-sort";
import jssIsolate from "jss-isolate";
import { JssProvider as JssProviderBase } from "react-jss";
import defaultUnits from "./defaultUnits";

const createGenerateClassName = () => (rule, sheet) =>
  `${sheet.options.componentName}-${rule.key}${sheet.options.index}`;

const jss = create({ createGenerateClassName });
jss.use(jssGlobal());
jss.use(jssExtend());
jss.use(jssNested());
jss.use(jssCamelCase());
jss.use(jssDefaultUnit(defaultUnits));
jss.use(jssExpand());
jss.use(jssVendorPrefixer());
jss.use(jssPropsSort());
jss.use(jssIsolate({ isolate: false }));

const JssProvider = ({ children }) => (
  <JssProviderBase jss={jss}>{children}</JssProviderBase>
);

JssProvider.propTypes = {
  children: node.isRequired
};
export default JssProvider;
