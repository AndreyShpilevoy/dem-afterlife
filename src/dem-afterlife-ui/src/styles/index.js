import React from "react";
import { node, string } from "prop-types";
import injectSheet, { ThemeProvider as ThemeProviderBase } from "react-jss";
import classNames from "classnames";
import defaultTheme from "./themes/default";
import overriddenTheme from "./themes/overridden";

const themeObjectName = {
  default: defaultTheme,
  overridden: overriddenTheme
};

const concatStyleNames = styleNameArray => classNames(...styleNameArray);

const ThemeProvider = ({ themeName, children }) => (
  <ThemeProviderBase theme={themeObjectName[themeName]}>
    {children}
  </ThemeProviderBase>
);
ThemeProvider.propTypes = {
  children: node.isRequired,
  themeName: string
};
ThemeProvider.defaultProps = {
  themeName: "default"
};
export { ThemeProvider, injectSheet, concatStyleNames };
