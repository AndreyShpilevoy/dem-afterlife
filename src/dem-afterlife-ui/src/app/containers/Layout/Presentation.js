import React from 'react';
import {string, node, shape} from 'prop-types';
import {sharedPropTypes} from 'utils';
import Notification from 'containers/Notification';
import Container from 'components/Container';
import Row from 'components/Row';
import Column from 'components/Column';
import {withStyles, ThemeProvider} from 'styles';
import Header from './components/Header';
import Footer from './components/Footer';
import globalStyles from './globalStyles';
import calculateStyles from './calculateStyles';

export const PresentationPure = ({children, themeName, navigationLinkArray, socialMediaLinkArray, styles}) => {
    const {contentWrapper, container, content} = styles;
    return <ThemeProvider name={themeName}>
        <div className={contentWrapper}>
            <Container className={container}>
                {/* title*/}
                <Notification />
                <Row>
                    <Header navigationLinkArray={navigationLinkArray}/>
                </Row>
                <Row className={content}>
                    <Column xs={12}>
                        {children}
                    </Column>
                </Row>
                <Row>
                    <Footer socialMediaLinkArray={socialMediaLinkArray}/>
                </Row>
            </Container>
        </div>
    </ThemeProvider>;
};

PresentationPure.propTypes = {
    styles: shape().isRequired,
    children: node,
    themeName: string,
    navigationLinkArray: sharedPropTypes.navigationLinkArray,
    socialMediaLinkArray: sharedPropTypes.socialMediaLinkArray
};

export default withStyles(theme => ({...globalStyles(), ...calculateStyles(theme)}))(PresentationPure);