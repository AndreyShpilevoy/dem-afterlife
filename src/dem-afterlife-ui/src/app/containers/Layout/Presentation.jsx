import React from 'react';
import {string, node, shape} from 'prop-types';
import {sharedPropTypes} from 'utils';
import Notification from 'containers/Notification';
import Breadcrumbs from 'containers/Breadcrumbs';
import Container from 'components/Container';
import Row from 'components/Row';
import Column from 'components/Column';
import {withStyles, ThemeProvider} from 'styles';
import Header from './components/Header';
import Footer from './components/Footer';
import globalStyles from './globalStyles';
import calculateStyles from './calculateStyles';

export const PresentationPure = ({children, themeName, navigationLinkArray, socialMediaLinkArray, breadcrumbArray, styles}) => {
    const {contentWrapper, container, content, breadcrumbs} = styles;
    return (<ThemeProvider name={themeName}>
        <div className={contentWrapper}>
            <Container className={container}>
                {/* title */}
                <Notification />
                <Row>
                    <Header navigationLinkArray={navigationLinkArray} />
                </Row>
                <Row className={breadcrumbs}>
                    <Column xs={12}>
                        <Breadcrumbs breadcrumbArray={breadcrumbArray} />
                    </Column>
                </Row>
                <Row className={content}>
                    <Column xs={12}>
                        {children}
                    </Column>
                </Row>
                <Row className={breadcrumbs}>
                    <Column xs={12}>
                        <Breadcrumbs breadcrumbArray={breadcrumbArray} />
                    </Column>
                </Row>
                <Row>
                    <Footer socialMediaLinkArray={socialMediaLinkArray} />
                </Row>
            </Container>
        </div>
    </ThemeProvider>);
};

PresentationPure.propTypes = {
    styles: shape().isRequired,
    children: node.isRequired,
    themeName: string,
    navigationLinkArray: sharedPropTypes.navigationLinkArray.isRequired,
    socialMediaLinkArray: sharedPropTypes.socialMediaLinkArray.isRequired,
    breadcrumbArray: sharedPropTypes.breadcrumbArray.isRequired
};

PresentationPure.defaultProps = {
    themeName: 'default'
};

export default withStyles(theme => ({...globalStyles(), ...calculateStyles(theme)}))(PresentationPure);