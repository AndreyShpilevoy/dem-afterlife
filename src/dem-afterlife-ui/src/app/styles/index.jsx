// import ThemedStyleSheet from 'react-with-classes/lib/ThemedStyleSheet';
// import {css, withStyles, ThemeProvider} from 'react-with-classes';
import React from 'react';
import {node, string} from 'prop-types';
import injectSheet, {ThemeProvider as ThemeProviderBase} from 'react-jss';
import classNames from 'classnames';
import JssProvider from './JssProvider';
import defaultTheme from './themes/default';
import overriddenTheme from './themes/overridden';

// ThemedStyleSheet.registerDefaultTheme(defaultTheme);
// ThemedStyleSheet.registerTheme('overridden', overriddenTheme);
// ThemedStyleSheet.registerInterface(jssInterface);
const themeObjectName = {
    default: defaultTheme,
    overridden: overriddenTheme
};

const concatStyleNames = styleNameArray => classNames(...styleNameArray);

const ThemeProvider = ({themeName, children}) => (
    <JssProvider>
        <ThemeProviderBase theme={themeObjectName[themeName]}>
            {children}
        </ThemeProviderBase>
    </JssProvider>
);
ThemeProvider.propTypes = {
    children: node.isRequired,
    themeName: string
};
ThemeProvider.defaultProps = {
    themeName: 'default'
};
export {ThemeProvider, injectSheet, concatStyleNames};