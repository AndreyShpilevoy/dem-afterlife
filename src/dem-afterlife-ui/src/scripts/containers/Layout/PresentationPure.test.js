/* eslint no-undef: 0, fp/no-unused-expression: 0, fp/no-nil: 0, fp/no-mutation:0 */

import React from 'react';
import {shallow} from 'enzyme';
import * as styles from 'styles';
import {PresentationPure} from './Presentation';

const ThemeProvider = () => <ThemeProvider/>;
styles.ThemeProvider = ThemeProvider;

describe('Layout Presentation Pure', () => {
    const stylesObject = {
        container: 'container-0-0',
        content: 'content-0-1',
        contentWrapper: 'contentWrapper-0-2'
    };
    it('component match expected snapshot', () => {
        expect(shallow(
        <PresentationPure
            themeName={'default'}
            navigationLinkArray={[]}
            socialMediaLinkArray={[]}
            styles={stylesObject}>
            <div>
                {'Layout Presentation content'}
            </div>
        </PresentationPure>
        )).toMatchSnapshot();
    });
});
