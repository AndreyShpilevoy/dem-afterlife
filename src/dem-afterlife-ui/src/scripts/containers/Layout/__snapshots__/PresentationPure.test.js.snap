// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Presentation Pure component match expected snapshot 1`] = `
<ThemeProvider
  name="default"
>
  <div
    className="contentWrapper-0-2"
  >
    <withStyles(ContainerPure)
      className="container-0-0"
    >
      <Connect(NotificationPure) />
      <withStyles(RowPure)>
        <withStyles(HeaderPure)
          navigationLinkArray={Array []}
        />
      </withStyles(RowPure)>
      <withStyles(RowPure)
        className="content-0-1"
      >
        <withStyles(ColumnPure)
          xs={12}
        >
          <div>
            Layout Presentation content
          </div>
        </withStyles(ColumnPure)>
      </withStyles(RowPure)>
      <withStyles(RowPure)>
        <withStyles(FooterPure)
          socialMediaLinkArray={Array []}
        />
      </withStyles(RowPure)>
    </withStyles(ContainerPure)>
  </div>
</ThemeProvider>
`;
